fpdf/text_region.py

```diff
@@ -446,6 +508,24 @@ def _render_column_lines(self, text_lines, top, bottom):
                 col_left, col_right = self.current_x_extents(self.pdf.y, 0)
                 if self.pdf.x < col_left or self.pdf.x >= col_right:
                     self.pdf.x = col_left
+                self.pdf.x += cur_paragraph.indent
+                if cur_bullet and not cur_bullet.rendered_flag:
+                    bullet_indent_shift = (
+                        cur_bullet.fragment.get_width() + cur_bullet.rel_x_displacement
+                    )
+                    self.pdf.x -= bullet_indent_shift
+                    self.pdf.y += cur_bullet.rel_y_displacement
+                    self.pdf._render_styled_text_line(
+                        cur_bullet.text_line,
+                        h=cur_bullet.text_line.height,
+                        border=0,
+                        new_x=XPos.LEFT,
+                        new_y=YPos.TOP,
```

Comment:
Will deal with this.

Reply:
Firefox just ignores the start value in your example, because it can only handle ints (as required by the HTML specs), which go up to a certain size. And when they're short enough, it aligns them on the right (as makes sense for ints), just next to the left margin of the text. The annoying thing is, that this may easily make the beginning of the number disappear off the screen to the left.
HTML is apparently also limited to use a single one out of a small selection of predefined characters for non-numeric bullets.

But converting HTML is only one of the many use cases for bulletet/numbered/indexed indented paragraphs.
We should not let that limit our options.

In the general case:

* a "bullet" can be any size character, or a sequence of any number of characters.
* and a "numbering item" is not restricted to integers, but can also include any number of characters and digits.

That does not mean that you have to implement every possible variety and combination right now. Just try to keep the possibility open that more features can be added later without changing the already existing parts of the API.

* different types of alignment for bullets/numbering, both horizontal and vertical
* if a long "bullet" or numbering string is left aligned and collides with the paragraph text, a possible option to move either of the two out of the way
* whatever I might not have thought of myself at the moment 🤷

Reply:
> Firefox just ignores the start value in your example, because it can only handle ints

Forgot that I did, indeed, use different code for that, with a smaller number. In any case, my code handles long bullets the same way as the browsers.

> But converting HTML is only one of the many use cases for bulletet/numbered/indexed indented paragraphs.
We should not let that limit our options

Aren't `Paragraph`s only used for handling HTML right now? I can't seem to find any current usages of `Paragraph` objects outside of when handling HTML.

> different types of alignment for bullets/numbering, both horizontal and vertical
if a long "bullet" or numbering string is left aligned and collides with the paragraph text, a possible option to move either of the two out of the way

Alright. In the case of bullets, how do we want to handle left-alignment when there is also non-top alignment?

Reply:
> Aren't Paragraphs only used for handling HTML right now?

Paragraphs are designed to be used by client code, as documented in the [text region basics](https://py-pdf.github.io/fpdf2/TextRegion.html). The diagram there shows how the different parts of the system play together.
That html.py can use them as well (and table.py will soon) is just a happy coincidence. But internal modules use it essentially in the same way that any normal user of the library would.

> Alright. In the case of bullets, how do we want to handle left-alignment when there is also non-top alignment?

You mean bullet text vertically sitting in the middle of a paragraph and overlapping the paragraph text there? That sounds like a fun challenge... I guess we'll cross that bridge when we get to it. Maybe we'll just have to disable any overlap clearing feature when the bullet is not top aligned.

I suggest you just keep it simple for now, and those advanced features may be added in the future. I've already made you change your PR a lot, so it's probably better to get the parts that you have now working and properly tested, and then merged.

Reply:
Apologies for bumping, but during the last attempt to resolve the pull request, `Black` complained about formatting of `html.py` (which, amusingly enough, was introduced by `Black` in the first place). I dealt with the issue the same day, but there don't seem to have been any further attempts to resolve the pull request. Is there a problem?